engines/director/lingo/lingo-builtins.cpp

```diff
@@ -2563,7 +2563,8 @@ void LB::b_puppetSprite(int nargs) {
 				// sprite in new frame before setting puppet (Majestic).
 				Channel *channel = sc->getChannelById(sprite.asInt());
 
-				channel->replaceSprite(sc->_frames[sc->getNextFrame()]->_sprites[sprite.asInt()]);
+				// TODO: Fix this properly.
```

Comment:
Maybe it is just setting the `_puppet` flag? Need more investigation, similar to `immdiateSpite`. These were implemented pretty early and could be all kind of obsolete.